homeassistant/components/habitica/__init__.py

```diff
@@ -46,6 +54,9 @@
         vol.Optional(CONF_SENSORS, default=list(SENSORS_TYPES)): vol.All(
             cv.ensure_list, vol.Unique(), [vol.In(list(SENSORS_TYPES))]
         ),
+        vol.Optional(CONF_TASKS, default=list(TASKS_TYPES)): vol.All(
```

Comment:
This is possibly conflicting with ADR3. See https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md

We prefer to allow users to disable entities instead and/or disable entities by default.

https://developers.home-assistant.io/docs/entity_registry_disabled_by

If this is not conflicting with ADR3 it's at least conflicting with ADR10.

We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Thanks you, @MartinHjelmare for constructive and helpful comments. I have just pushed the requested changes :) :+1: